test/saithrift/src/switch_sai_rpc_server.cpp

```diff
@@ -2269,6 +2286,145 @@ class switch_sai_rpcHandler : virtual public switch_sai_rpcIf {
       status = wred_api->remove_wred_profile((sai_object_id_t) wred_id);
       return status;
   }
+  
+  void sai_thrift_parse_tunnel_attributes(const std::vector<sai_thrift_attribute_t> & thrift_attr_list,sai_attribute_t *attr_list){
+
+      std::vector<sai_thrift_attribute_t>::const_iterator it = thrift_attr_list.begin();
+      sai_thrift_attribute_t attribute;
+      for(uint32_t i = 0; i < thrift_attr_list.size(); i++, it++) {
+          attribute = (sai_thrift_attribute_t)*it;
+          attr_list[i].id = attribute.id;
+          switch (attribute.id) {
+                case SAI_TUNNEL_ATTR_TYPE:
+                        attr_list[i].value.s32 = attribute.value.s32;
+                        break;
+                case SAI_TUNNEL_ATTR_UNDERLAY_INTERFACE :
+                        attr_list[i].value.oid = attribute.value.oid;
+                        break;
+                case SAI_TUNNEL_ATTR_OVERLAY_INTERFACE:
+                        attr_list[i].value.oid =attribute.value.oid;
+                        break;
+                case SAI_TUNNEL_ATTR_SRC_IP:
+                        sai_thrift_parse_ip_address(attribute.value.ipaddr, &attr_list[i].value.ipaddr);
+                        break;
+                case SAI_TUNNEL_ATTR_ENCAP_TTL_MODE :
+                        attr_list[i].value.u32=attribute.value.u32;
+                        break;
+                case SAI_TUNNEL_ATTR_ENCAP_DSCP_MODE :
+                        attr_list[i].value.u32=attribute.value.u32;
+                        break;
+                case SAI_TUNNEL_ATTR_ENCAP_TTL_VAL :
+                        attr_list[i].value.u8=attribute.value.u8;
+                        break;
+                case SAI_TUNNEL_ATTR_ENCAP_DSCP_VAL : 
+                        attr_list[i].value.u8 = attribute.value.u8;
+                        break;
+                case SAI_TUNNEL_ATTR_DECAP_TTL_MODE :
+                        attr_list[i].value.u32=attribute.value.u32;
+                        break;
+                case SAI_TUNNEL_ATTR_DECAP_DSCP_MODE: 
+                        attr_list[i].value.u32 = attribute.value.u32;
+                        break;
```

Comment:
We went with the required attributes that helps for making the use cases that we created in saitunnel file.We didnot cover possible number of attributes here.we will try to cover